src/core/http/serializable_header_map.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Debug)]
+pub struct SerializableHeaderMap(HeaderMap);
+
+impl SerializableHeaderMap {
+    pub fn new(headers: HeaderMap) -> Self {
+        Self(headers)
+    }
+
+    pub fn headers(&self) -> HeaderMap {
+        self.0.clone()
+    }
+}
+
+impl<'de> Deserialize<'de> for SerializableHeaderMap {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        // Deserialize a HashMap<String, String>
+        let map: HashMap<String, String> = Deserialize::deserialize(deserializer)?;
+
+        // Convert the HashMap<String, String> to a HeaderMap
+        let mut header_map = HeaderMap::new();
+        for (key, value) in map {
+            header_map.insert(
+                HeaderName::from_str(&key).map_err(serde::de::Error::custom)?,
+                HeaderValue::from_str(&value).map_err(serde::de::Error::custom)?,
+            );
+        }
```

Comment:
can we use https://docs.rs/reqwest/latest/reqwest/header/struct.HeaderMap.html#impl-TryFrom%3C%26HashMap%3CK,+V%3E%3E-for-HeaderMap%3CT%3E ?